boa/src/builtins/object/mod.rs

```diff
@@ -410,6 +410,24 @@ pub fn make_object(_: &Value, args: &[Value], ctx: &mut Interpreter) -> ResultVa
     Ok(object)
 }
 
+/// Creates a new object from the provided prototype
+///
+/// https://tc39.es/ecma262/#sec-object.create
+/// https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/create
+pub fn create_builtin(_: &Value, args: &[Value], interpreter: &mut Interpreter) -> ResultValue {
+    let __proto__ = args.get(0).cloned().unwrap_or_else(Value::undefined);
+    match __proto__ {
+        Value::Object(_) | Value::Null => Ok(Value::new_object_from_prototype(
+            __proto__,
+            ObjectData::Ordinary,
+        )),
+        _ => interpreter.throw_type_error(format!(
+            "Object prototype may only be an Object or null: {}",
+            __proto__
+        )),
+    }
+}
```

Comment:
We should rename this to `prototype`, rather than it being `__proto__`.

Even in the standard it says:
> The create function creates a new object with a specified prototype. When the create function is called, the following steps are taken:

Reply:
Both `prototype` and `__proto__` point to the "prototype". 

`prototype` is the property of the creator function (so used in that context), while `__proto__` is the property of the "child object" (so used in that context).  

That's why I think `__proto__` is better here.

Reply:
> Both `prototype` and `__proto__` point to the "prototype".
> 
> `prototype` is the property of the creator function (so used in that context), while `__proto__` is the property of the "child object" (so used in that context).
> 
> That's why I think `__proto__` is better here.

I know both point to the same thing, but It's more semantically meaningful when you call `Object.create(String.prototype)` we create an object from the prototype, rather than the prototype instance.

Reply:
Maybe I missed the point of your last comment. But you wouldn't call `Object.create`  with the functions prototype. Instead you would do `new SomeFunction ()`. I think you call `Object.create` directly with the  __proto__ instance.

Reply:
> Maybe I missed the point of your last comment, but you wouldn't call `Object.create` with the function's prototype. Instead you would do `new SomeFunction ()`. I think in vast majority of cases you call `Object.create` directly with the `__proto__` instance.

Sorry, I'll try to explain this a bit better, I'll take an examle from mdn:
```js
const person = {
  isHuman: false,
  printIntroduction: function() {
    console.log(`My name is ${this.name}. Am I human? ${this.isHuman}`);
  }
};

const me = Object.create(person);
```
In this case we create a new object with the `prototype` (`person`), and when the object is created the `[[prototype]]` (prototype instance) of the new object is set to `prototype`, it's set in `step. 2` in <https://tc39.es/ecma262/#sec-object.create> which in boa is `Value::new_object_from_prototype()` in creates the instance of the `prototype` (`__proto__`)

Reply:
OK, but it is not clear to me that naming it `prototype` locally in this function is any better than `__proto__`. Even in https://tc39.es/ecma262/#sec-ordinaryobjectcreate the parameter is called _proto_. So this might even be a bit better. (Although in the function we are talking about they call it as `O`). 

The function we call from this point `pub fn new_object_from_prototype(proto: Value, data: ObjectData)` also uses the argument `proto`, as in the spec. (although the name of this function is not aligned with the spec's one).

Reply:
> OK, but it is not clear to me that naming it `prototype` locally in this function is any better than `__proto__`. Even in [tc39.es/ecma262/#sec-ordinaryobjectcreate](https://tc39.es/ecma262/#sec-ordinaryobjectcreate) the parameter is called _proto_. So this might even be a bit better. (Although in the function we are talking about they name this parameter as `O`).

Yes. the spec is confusing at times, I think it's `O` (as in object) because the prototoype can be any object (like the example above with `person` object) and technically `null` is an object (`typeof null == "object"`)

You can know it's an instance if there is `[[Prototype]]`, otherwise assume it's `prototype` (the spec is not always consistent).

Some good informative blogs are the V8 `Understanding the ECMAScript spec` there are currently 4 parts here is part one https://v8.dev/blog/understanding-ecmascript-part-1

Reply:
So you still think it is important to rename this local variable from `__proto__` to `prototype`? If so, I'll change it.

Reply:
> So you still think it is important to rename this local variable from `__proto__` to `prototype`? If so, I'll change it.

Yes. It makes more sense to name it `prototype`.